### PR TITLE
Cut 0.23.0 release.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 
 ## Unreleased
 
-## [v0.23.0-rc.0](https://github.com/thanos-io/thanos/tree/release-0.23) - 2021.09.02
+## [v0.23.0](https://github.com/thanos-io/thanos/tree/release-0.23) - 2021.09-16
```
Comment (kakkoyun) [inline CHANGELOG.md:13]:
```suggestion
## [v0.23.0](https://github.com/thanos-io/thanos/tree/release-0.23) - 2021.09.16
```

**Review by kakkoyun — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (bwplotka) [reply CHANGELOG.md]: Fixed, thanks!

**Comment by yeya24** [comment]
Can we have https://github.com/thanos-io/thanos/pull/4657 as well?

**Comment by yeya24** [comment]
I'd like to have https://github.com/thanos-io/thanos/pull/4681 to get into this release as well. With the `EndpointSet` change introduced in 0.23.0, federated proxy APIs are enabled automatically so #4681 is definitely necessary to have.

**Comment by bwplotka** [comment]
Thanks for pointing, let me look

**Comment by bwplotka** [comment]
Closing for now, let's do 0.23.0-rc.1 first.

**Pull request closed without merging**